org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java

```diff
@@ -217,10 +221,20 @@ public static boolean isOffsetInRange(int offset, Range range, IDocument documen
 		}
 	}
 
-	public static CompletionParams toCompletionParams(URI fileUri, int offset, IDocument document)
+	public static CompletionParams toCompletionParams(URI fileUri, int offset, IDocument document, char[] completionTriggerChars)
 			throws BadLocationException {
 		Position start = toPosition(offset, document);
 		final var param = new CompletionParams();
+		try {
+			String positionCharacter = document.get(offset-1, 1);
+			if (Chars.contains(completionTriggerChars, positionCharacter.toCharArray()[0])) {
+				param.setContext(new CompletionContext(CompletionTriggerKind.TriggerCharacter, positionCharacter));
+			} else {
+				param.setContext(new CompletionContext(CompletionTriggerKind.Invoked, positionCharacter));
```

Comment:
According to https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionContext I think it should rather be param.setContext(new CompletionContext(CompletionTriggerKind.Invoked));

As it says
```
	/**
	 * The trigger character (a single character) that has trigger code
	 * complete. Is undefined if
	 * `triggerKind !== CompletionTriggerKind.TriggerCharacter`
	 */
```
What do you think?

Reply:
I would propose:
```java
} else {
  // According to LSP 3.17 specification: the triggerCharacter in CompletionContext is undefined if
  // triggerKind != CompletionTriggerKind.TriggerCharacter
  param.setContext(new CompletionContext(CompletionTriggerKind.Invoked, positionCharacter));
}
```